force-app/main/default/classes/FlowControlLoopsCollections.cls

```diff
@@ -109,48 +143,56 @@ public with sharing class FlowControlLoopsCollections {
      * @param score The score.
      * @return The grade for the given score.
      */
-    public static String determineGrade(Integer score) {
-        return null; // Replace null with the variable you used to store the result
-    }
-
+        public static String determineGrade(Integer score) {
+        if (score >= 90) {
+            return 'A';
+        } else if (score >= 80) {
+            return 'B';
+        } else if (score >= 70) {
+            return 'C';
+        } else if (score >= 60) {
+            return 'D';
+        } else {
+            return 'F';
+        }
+        }
 
-    /**
+    /** Q8
      * Question 8
      * Sum all the integers up to a given limit.
      * Example: sumUpToLimit(5) should return 15
      * @param intLimit The number up to which integers will be summed.
      * @return The sum of all integers up to the limit.
      */
-    public static Integer sumUpToLimit(Integer intLimit) {
-        /* 
-         for (){
-            
-         }
-        */
-
-        return null; // Replace null with the variable you used to store the result
-    }
+        public static Integer sumUpToLimit(Integer intLimit) {
+        Integer sum = 0;
+             for (Integer i = 1; i <= intLimit; i++){
+               sum += i; 
+             }
+            return sum;
+        }
 
-    /**
+    /** Q9
      * Question 9
      * Generate String "Hello World!" three times, with each occurrence separated by a semicolon.
      * Example: returnHelloWorld() should return "Hello World!; Hello World!; Hello World!; "
      * @return The string "Hello World!; Hello World!; Hello World!; "
      */
-    public static String returnHelloWorld() {
-        // Initialize the result String
+        public static String returnHelloWorld() {
+            // Initialize the result String
+            String result = '';
 
-        // Use a for loop to append 'Hello World!;' to the result string three times
-        /* 
-         for (){
+            // Use a for loop to append 'Hello World!;' to the result string three times
             
-         }
-        */
+            for (Integer i = 0; i< 3; i++) {
+                result += String.valueOf('Hello World!') + '; ';
```

Comment:
I'm curious why you decided to use **String.valueOf()** here. `'Hello Word!'` is already a String -- so no need to convert it to a String:
```suggestion
                result += 'Hello World!; ';
```
Let me know if you have questions!